src/launch.rs

```diff
@@ -16,9 +16,8 @@ impl TryFrom<Procfile> for Launch {
             launch.processes.push(Process {
                 r#type: ProcessType::from_str(&key)
                     .map_err(ProcfileConversionError::InvalidProcessType)?,
-                command: value,
+                command: vec![String::from("bash"), String::from("-c"), value],
                 args: Vec::<String>::new(),
```

Comment:
A big design decision we need to make here, is whether to put `value` as the last element of the `command`, or as the only element of `args`. (This came up in #150 but I never got as far as investigating before closing the PR out for the cnb-shim reasons.)

If it's in `command`, then I would guess `docker run --rm -it myimage <foo>` would result in the approx equivalent of:
`bash -c "<value>" "<foo>"`

...and I'm not sure what bash would do in that case (since `<foo>` isn't part of the arg associated with the `-c`).

Plus I think it would prevent eg `docker run --rm -it myimage echo "hello"` from working?

If instead we put `value` in `args`, then users couldn't append single args (though given the above issue about `-c` maybe that doesn't work anyway), but at least would get a clean slate when they try using it like they would `heroku run bash` or `heroku run ./manage.py migrate` etc?

It also doesn't help that https://buildpacks.io/docs/app-developer-guide/run-an-app/#run-a-multi-process-app still hasn't been updated for the bash/`direct=false` removal, otherwise that might provide some insight.

Reply:
> I'm not sure what bash would do in that case (since `<foo>` isn't part of the arg associated with the `-c`).

Just with some weak local testing, the `<foo>` seems to get lost.

```bash
$ bash -c "echo 'hello'" "where does this go"
hello
$ bash -c "echo" "hello"


```

> Plus I think it would prevent eg `docker run --rm -it myimage echo "hello"` from working?

I don't think this will work in any scenario. I think this would result in

with `value` as `command[2]`: `bash -c "mywebserver" "echo" "hello"`
with `value` as `args[0]`: `bash -c "echo" "hello"`

Neither of the above would pass "hello" as an arg to "echo". If the whole command is quoted (`docker run --rm -it myimage "echo 'hello'"`) the `args[0]` option *should* work.

No matter which solution we pick, users providing additional arguments will end up going to bash, which will end up being ignored. This isn't a great UX, but I'm not sure there is a better option without writing our own launcher or some questionable bash.

The `args[0]` approach is the most like `heroku run` today (allowing folks to run bash in a web process), so I think I prefer that.

Reply:
Documenting for posterity:

- Passing a single command (eg `docker run myimage bash`) -> now works (when it didn't before)
- Passing a command and args **unquoted** (eg `docker run myimage echo hello`) -> doesn't work (unchanged from before)
- Passing a command and args **quoted** (eg `docker run myimage "echo hello"`) -> now works (when it didn't before)
- Passing a command and args but also overriding the entrypoint (eg `docker run --entrypoint launcher myimage echo hello`) -> works (unchanged from before)